src/m365/pp/commands/solution/solution-get.spec.ts

```diff
@@ -0,0 +1,297 @@
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { powerPlatform } from '../../../../utils/powerPlatform';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./solution-get');
+
+describe(commands.SOLUTION_GET, () => {
+  let commandInfo: CommandInfo;
+  const validEnvironment = '4be50206-9576-4237-8b17-38d8aadfaa36';
+  const validId = 'ee62fd63-e49e-4c09-80de-8fae1b9a427e';
+  const validName = 'Default';
+  const invalidId = 'Invalid GUID';
+  const envUrl: any = "https://contoso-dev.api.crm4.dynamics.com";
+  const solutionResponse: any = {
+    "value": [
+      {
+        "solutionid": "00000001-0000-0000-0001-00000000009b",
+        "uniquename": "Crc00f1",
+        "version": "1.0.0.0",
+        "installedon": "2021-10-01T21:54:14Z",
+        "solutionpackageversion": null,
+        "friendlyname": "Common Data Services Default Solution",
+        "versionnumber": 860052,
+        "publisherid": {
+          "friendlyname": "CDS Default Publisher",
+          "publisherid": "00000001-0000-0000-0000-00000000005a"
+        }
+      }
+    ]
+  };
+  const solutionResponseText: any = {
+    "uniquename": "Crc00f1",
+    "version": "1.0.0.0",
+    "publisher": "CDS Default Publisher"
+  };
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      powerPlatform.getDynamicsInstanceApiUrl
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.SOLUTION_GET), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['uniquename', 'version', 'publisher']);
+  });
+
+  it('fails validation when no solution found', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').returns(envUrl);
```

Comment:
getDynamicsInstanceApiUrl returns a promise, so we should either do
`.returns(Promise.resolve(envUrl)`
or
`.callsFake(async () => envUrl);`

I prefer the latter one.